src/odemis/gui/cont/streams.py

```diff
@@ -386,6 +386,25 @@ def add_metadata(self, key, value, unit=None):
         except Exception:
             logging.exception("Trying to convert metadata %s", key)
 
+    def add_txt_entry(self, panel, lbl, txt_ctrl, ftsize=None):
+        # wx.StaticText isn't displayed for some unknown reason, use TextCtrl instead
```

Comment:
How much of the code could be replaced by `self.stream_panel.add_text_field()`? Also, in general, I'd avoid passing `panel`, as it should be fixed to the panel of the stream controller (self.stream_panel).